### DISCUSSION TITLE
Support jitter to the alert scheduler module

### DISCUSSION THREAD

**Discussion by chinmaygupta28** [grafana/grafana#47861] [Alerting]
We face a lot of throttling from the Cloudwatch datasource on the alerts that we have put onto Grafana for the Cloudwatch metrics (we have a lot of these). Looking at the Usage count metrics that AWS provides we are always under the limit when we look into the aggregations over a minute but the other metric that AWS provides - ThrottleCount is very high. Such graphs only suggests that Grafana must be scheduling these at some second mark like 0th second, 15th second, 30th second etc. 

I wanted to check for an evaluation period of the 60s can the scheduled time of each of the evaluation calculations be done with a configurable jitter amount of time? I would like to have a configuration where if one query was made to the cloudwatch at T timestamp then the next one can happen between T+30 to T+90 seconds so that the requests are evenly distributed for our upstream datasource.

Let me know if there are ideas to tackle these.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
